exporter/splunkhecexporter/client_test.go

```diff
@@ -1362,6 +1362,76 @@ func TestInvalidURL(t *testing.T) {
 	assert.EqualError(t, err, "Post \"ftp://example.com:134/services/collector\": unsupported protocol scheme \"ftp\"")
 }
 
+func TestHeartbeatStartupFailed(t *testing.T) {
```

Comment:
Can you add another test that's practically identical to this, but make sure everything succeeds if `cfg.Heartbeat.Startup = false`, even if the heartbeat would fail?